tests/network/libs/connectivity.py

```
@@ -33,6 +34,28 @@ def build_ping_command(dst_ip: str, count: int, timeout: int) -> str:
+    Args:
+        ping_output: The full output of a ping command.
+
+    Returns:
+        The packet-loss percentage (0-100) reported in the statistics line.
+
+    Raises:
+        PacketLossSummaryNotFoundError: If no packet-loss summary line is found in the output.
+    """
+    match = re.search(pattern=r"(\d+(?:\.\d+)?)% packet loss", string=ping_output)
+    if not match:
+        raise PacketLossSummaryNotFoundError(f"No packet-loss summary found in ping output: {ping_output}")
```

Ruff (0.16.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)